ci/logs.py

```diff
@@ -106,17 +106,26 @@ def get_and_zip_logs(
                 if tail_lines:
                     args['tail_lines'] = tail_lines
 
-                data, status_code, headers = k8s.read_namespaced_pod_log_with_http_info(**args)
-                # returns tuple (response_data, http status, headers)
-                log_response = K8sResponse(
-                    data=data,
-                    status_code=status_code,
-                    headers=headers,
-                )
-
-                if log_response.status_code != 200:
-                    print(f'Getting logs failed with {log_response.status_code=}')
-                    continue
+                try:
+                    data, status_code, headers = k8s.read_namespaced_pod_log_with_http_info(**args)
+                    # returns tuple (response_data, http status, headers)
+                    log_response = K8sResponse(
+                        data=data,
+                        status_code=status_code,
+                        headers=headers,
+                    )
+
+                    if log_response.status_code != 200:
```

Comment:
did not check response's type. if it is `requests`-style, you might check `respons.ok`.
generally speaking, all status-codes starting w/ 2xx should be considered okay..